### PR TITLE
added iam user

### PR DESCRIPTION
**Pull request by rhodgman-r7**
added template resource representing iam user
updated iam policy template resource to handle users (and groups)

### CODE REVIEW

File: example/iam_user.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        :Action => ['s3:GetObject'],
+        :Resource => ['arn:aws:s3:::some.bucket.name.here/*']
+      }]
+    )
+  end
+
+  iam_role 'NewUser' do
+    path fn_ref(:Path)
+  end
+end
```
Comment (athompson-r7) [inline example/iam_user.rb]: Should this be a call to `iam_user`, because you are calling `user` in the policy?

Comment (rhodgman-r7) [inline example/iam_user.rb]: Yup, nice catch, I'll fix that up soon.

**Comment by rhodgman-r7** [comment]
I updated the example, this should be good to know now. Also, at some point we may want to go through and make these examples a little more flushed out, using some more of the available properties and methods.

**Comment by jmanero-r7** [comment]
:+1:

**Pull request merged**